src/elastic-search/Elasticsearch_guide.md

```diff
@@ -0,0 +1,211 @@
+# Getting started
+
+To be able to start application with Elasticsearch you will need to:
+
+1. Start with `npm prepare:local` to executes docker-compose file which pull and run Elasticsearch cluster as a docker container.
+2. Start the application with `npm run start`
+
+# Check the .env file
+
+Following environment variables are required to run Elasticsearch with Scicat Backend:
+
+| `Variables`              | Values                 | Description                                                                                                    |
+| ------------------------ | :--------------------- | :------------------------------------------------------------------------------------------------------------- |
+| `ELASTICSEARCH_ENABLED`  | `yes` or `no`          | Flag to enable/disable the ElasticSearch service                                                               |
+| `ES_HOST `               | https://localhost:9200 | Use `http` if `xpack.security.enabled` set to `false`                                                          |
+| `ES_USERNAME` (optional) | elastic                | Elasticsearch cluster username. Can be customized with docker image environment values with `ELASTIC_USERNAME` |
+| `ES_PASSWORD`            | password               | Elasticsearch cluster password. Can be customized with docker image environment values `ELASTIC_PASSWORD`      |
+| `MONGODB_COLLECTION `    | Dataset                | Collection name to be mapped into specified Elasticsearch index                                                |
+| `ES_MAX_RESULT    `      | 210000                 | Maximum records can be indexed into Elasticsearch. If not set, default is `10000`                              |
+| `ES_FIELDS_LIMIT `       | 400000                 | The total number of fields in an index. If not set, default is `1000`                                          |
+| `ES_INDEX     `          | dataset                | Setting default index for the application                                                                      |
+| `ES_REFRESH  `           | wait_for               | Wait for the change to become visible - CRUD actions                                                           |
+|                          |                        |
+
+## \*\* NOTE \*\*
+
+`ES_REFRESH="wait_for"` is used for better testing flow. For production it should be set to false. Detailed docs about this setting can be found: [docs-refresh](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html)
+
+# Steps to synchronize data with Elasitcsearch
+
+To enable search queries in Elasticsearch, you should adhere to the following steps:
+
```

Comment:
This is the manual workflow, needs probably some more info on how to run that on production instances

Reply:
I will check with @Junjiequan regarding this. If he has not implemented this yet, can we postpone it to a followup PR?

Reply:
Yes